csvqb/csvqb/configloaders/infojson1point1/columnschema.py

```diff
@@ -0,0 +1,459 @@
+    isRequired: bool = False
+    value: Optional[str] = None
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "NewAttribute":
+        instance = _from_dict(cls, d)
+        if isinstance(instance.new, dict):
+            instance.new = NewAttributeProperty.from_dict(instance.new)  # type: ignore
+        return instance
+
+    def map_to_new_qb_attribute(
+        self, column_title: str, data: PandasDataTypes
+    ) -> NewQbAttribute:
+        if isinstance(self.new, bool) and self.new:
+            return NewQbAttribute.from_data(
+                label=column_title, data=data, is_required=self.isRequired
+            )
+        elif isinstance(self.new, NewAttributeProperty):
+            return NewQbAttribute(
+                label=self.new.label or column_title,
+                description=self.new.comment,
+                new_attribute_values=_get_new_attribute_values(
+                    data, self.new.newAttributeValues
+                ),
+                parent_attribute_uri=self.new.subPropertyOf,
+                source_uri=self.new.isDefinedBy,
+                is_required=self.isRequired,
+                uri_safe_identifier_override=self.new.path,
+            )
+        else:
+            raise ValueError(f"Unhandled 'new' value: {self.new}")
+
+
+@dataclass
+class ExistingAttribute(SchemaBaseClass):
+    uri: str
+    value: Optional[str] = None
+    newAttributeValues: Union[None, bool, List[NewAttributeValue]] = None
+    isRequired: bool = False
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "ExistingAttribute":
+        instance = _from_dict(cls, d)
+        if isinstance(instance.newAttributeValues, list):
+            instance.newAttributeValues = [
+                NewAttributeValue.from_dict(attr_val)  # type: ignore
+                for attr_val in instance.newAttributeValues
+            ]
+        return instance
+
+    def map_to_existing_qb_attribute(
+        self, data: PandasDataTypes
+    ) -> ExistingQbAttribute:
+        return ExistingQbAttribute(
+            self.uri,
+            new_attribute_values=_get_new_attribute_values(
+                data, self.newAttributeValues
+            ),
+            is_required=self.isRequired,
+        )
+
+
+@dataclass
+class NewMeasure(SchemaBaseClass):
+    label: str
+    path: Optional[str] = None
+    comment: Optional[str] = None
+    isDefinedBy: Optional[str] = None
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "NewMeasure":
+        return _from_dict(cls, d)
+
+
+@dataclass
+class NewUnit(SchemaBaseClass):
+    label: str
+    path: Optional[str] = None
+    comment: Optional[str] = None
+    isDefinedBy: Optional[str] = None
+    baseUnit: Optional[str] = None
+    baseUnitScalingFactor: Optional[float] = None
+    qudtQuantityKind: Optional[str] = None
+    siBaseUnitConversionMultiplier: Optional[float] = None
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "NewUnit":
+        return _from_dict(cls, d)
+
```

Comment:
Apropos of line 265, I cannot imagine the pain to not use pydantic rn.

Reply:
Indeed, dataclasses make the whole job of deserialising from JSON much simpler.